OpenRA.Mods.Common/Traits/World/PaletteFromGimpFile.cs

```diff
@@ -0,0 +1,110 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Load a GIMP .gpl Palette file.")]
+	class PaletteFromGimpFileInfo : ITraitInfo
+	{
+		[FieldLoader.Require, PaletteDefinition]
+		[Desc("Palette name used internally.")]
+		public readonly string Name = null;
+
+		[Desc("If defined, load the palette only for this tileset.")]
+		public readonly string Tileset = null;
+
+		[FieldLoader.Require]
+		[Desc("Name of the file to load.")]
+		public readonly string Filename = null;
+
+		[Desc("Map listed indices to shadow. Ignores previous color.")]
+		public readonly int[] ShadowIndex = { };
```

Comment:
This doesnt work -- you should be passing it to the ImmutablePalette ctor.

Reply:
Thanks. I noticed during testing, but wasn't sure if the file was butchered. Should be fixed now.